Makefile

```diff
@@ -99,7 +104,28 @@ coverdir:
 	mkdir -p /tmp/tigris_coverdata && chmod a+w /tmp/tigris_coverdata; rm -f /tmp/tigris_coverdata/*
 	mkdir -p /tmp/tigris_coverdata2 && chmod a+w /tmp/tigris_coverdata2; rm -f /tmp/tigris_coverdata2/*
 
-generate: ${GEN_DIR}/api.pb.go ${GEN_DIR}/api.pb.gw.go ${GEN_DIR}/health.pb.go ${GEN_DIR}/health.pb.gw.go ${GEN_DIR}/admin.pb.go ${GEN_DIR}/admin.pb.gw.go ${DATA_PROTO_DIR}/data.pb.go
+
+generate: generate_api generate_mocks
+
+generate_api: ${GEN_DIR}/api.pb.go ${GEN_DIR}/api.pb.gw.go ${GEN_DIR}/health.pb.go ${GEN_DIR}/health.pb.gw.go ${GEN_DIR}/admin.pb.go ${GEN_DIR}/admin.pb.gw.go ${DATA_PROTO_DIR}/data.pb.go
+
+# generate mocks
+generate_mocks:
+	for ifaces in $(MOCK_INTERFACES) ; do \
+		MOCK_DIR=$${ifaces%:*} ; \
+		MOCK_NAME="$${ifaces#*:}" ; \
+		mockery --dir $${MOCK_DIR} --name $${MOCK_NAME} ; \
+  	done
+
+# clean mocks
+clean_mocks:
+	for ifaces in $(MOCK_INTERFACES) ; do \
```

Comment:
That's why I am saying that having all the mock in one top level package would be easier.
Clean would be just rm -f mock/*
In generate MOCK_DIR also would be static.

Reply:
I am running into circular imports that way if interfaceB uses interfaceA:

```
mocks/
    mock_interfaceB.go   // imports packageA/packageB
packageA/
    test_interfaceA.go.  // imports packageB && mocks
    packageB/
         interfaceB.go
         others.go
```